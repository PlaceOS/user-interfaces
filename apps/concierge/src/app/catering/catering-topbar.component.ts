import { Component, OnInit } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';
import { first, map, take } from 'rxjs/operators';

import { AsyncHandler, SettingsService } from '@placeos/common';
import { OrganisationService } from '@placeos/organisation';
import {
    CateringOrdersService,
    CateringStateService,
    ChargeCodeListModalComponent,
} from '@placeos/catering';
import { MatDialog } from '@angular/material/dialog';
import { AvailableRoomsStateModalComponent } from '@placeos/components';
import { combineLatest } from 'rxjs';

@Component({
    selector: 'catering-topbar',
    template: `
        <div class="flex items-center w-full pt-4 pb-2 px-8 space-x-2">
            <h2 class="text-2xl font-medium">
                Catering {{ page === 'menu' ? 'Menu' : 'Orders' }}
            </h2>
            <div class="flex-1 w-px"></div>
            <searchbar
                class="mr-2"
                [model]="filters?.search"
                (modelChange)="setSearch($event)"
            ></searchbar>
        </div>
        <div class="flex items-center bg-base-100 h-20 px-8 space-x-2">
            <mat-form-field appearance="outline" class="no-subscript w-60">
                <mat-select
                    [ngModel]="filters?.zones"
                    (ngModelChange)="updateZones($event)"
                    placeholder="All Levels"
                    multiple
                >
                    <mat-option
                        *ngFor="let level of levels | async"
                        [value]="level.id"
                    >
                        <div class="flex flex-col-reverse">
                            <div class="text-xs opacity-30" *ngIf="use_region">
                                {{ (level.parent_id | building)?.display_name }}
                                <span class="opacity-0"> - </span>
                            </div>
                            <div>
                                {{ level.display_name || level.name }}
                            </div>
                        </div>
                    </mat-option>
                </mat-select>
            </mat-form-field>
            <div *ngIf="page === 'menu'" class="flex-1 w-2"></div>
            <button
                *ngIf="
                    page === 'menu' && (!zones[0] || zones[0] === building?.id)
                "
                icon
                matRipple
                matTooltip="Add Item"
                class="bg-secondary text-secondary-content rounded h-12 w-12"
                (click)="addItem()"
            >
                <app-icon class="text-2xl">add</app-icon>
            </button>
            <button
                *ngIf="page === 'menu'"
                icon
                matRipple
                matTooltip="Edit Config"
                class="bg-secondary text-secondary-content rounded h-12 w-12"
                (click)="editConfig()"
            >
                <app-icon class="text-2xl">menu_book</app-icon>
            </button>
            <button
                *ngIf="page === 'menu'"
                icon
                matRipple
                matTooltip="Import Menu"
                class="bg-secondary text-secondary-content rounded h-12 w-12"
                (click)="importMenu()"
            >
                <app-icon class="text-2xl">cloud_upload</app-icon>
            </button>
            <button
                *ngIf="page === 'menu'"
                icon
                matRipple
                matTooltip="Room Availability"
                class="bg-secondary text-secondary-content rounded h-12 w-12"
                (click)="setRoomAvailability()"
            >
                <app-icon class="text-2xl">event_available</app-icon>
            </button>
            <button
                *ngIf="page === 'menu'"
                icon
                matRipple
                matTooltip="Charge Codes"
                class="bg-secondary text-secondary-content rounded h-12 w-12"
                (click)="setChargeCodes()"
            >
                <app-icon class="text-2xl">payments</app-icon>
            </button>
            <div *ngIf="page !== 'menu'" class="flex-1 w-2"></div>
            <!-- <searchbar class="mr-2"></searchbar> -->
            <date-options
                *ngIf="page !== 'menu'"
                (dateChange)="setDate($event)"
            ></date-options>
        </div>
    `,
    styles: [
        `
            mat-form-field {
                height: 3.25em;
                width: 8em;
            }
        `,
    ],
})
export class CateringTopbarComponent extends AsyncHandler implements OnInit {
    /** List of selected levels */
    public zones: string[] = [];
    /** Currently active page */
    public page: string;
    public readonly filters = this._orders.filters;
    /** List of levels for the active building */
    public readonly levels = combineLatest([
        this._org.active_building,
        this._org.active_region,
    ]).pipe(
        map(([bld, region]) =>
            this._settings.get('app.use_region')
                ? this._org.levelsForRegion(region)
                : this._org.levelsForBuilding(bld)
        )
    );
    /** Set filtered date */
    public readonly setDate = (date) =>
        (this._orders.filters = { ...this._orders.filters, date });
    public readonly setSearch = (str) =>
        (this._orders.filters = { ...this._orders.filters, search: str });
    /** List of levels for the active building */
    public readonly updateZones = (z) => {
        this._router.navigate([], {
            relativeTo: this._route,
            queryParams: { zone_ids: z.join(',') },
        });
        this._orders.filters = { ...this._orders.filters, zones: [z] };
        this._catering.zone = z[0];
    };

    public readonly addItem = () => this._catering.addItem();
    public readonly editConfig = () => this._catering.editConfig();
    public readonly importMenu = () => this._catering.importMenu();

    public get building() {
        return this._org.building;
    }

    public get use_region() {
        return !!this._settings.get('app.use_region');
    }

    constructor(
        private _orders: CateringOrdersService,
        private _catering: CateringStateService,
        private _org: OrganisationService,
        private _route: ActivatedRoute,
        private _router: Router,
        private _dialog: MatDialog,
        private _settings: SettingsService
    ) {
        super();
    }

    public async ngOnInit() {
        await this._org.initialised.pipe(first((_) => _)).toPromise();
        this._catering.zone =
            (this._orders.filters?.zones || [])[0] || this._org.building?.id;
        this.subscription(
            'route.query',
            this._route.queryParamMap.subscribe((params) => {
                if (params.has('zone_ids')) {
                    const zones = params.get('zone_ids').split(',');
                    if (!zones.length) return;
                    const level = this._org.levelWithID(zones);
                    this.zones = zones;
                    if (!level) return;
                    this._org.building = this._org.buildings.find(
                        (bld) => bld.id === level.parent_id
                    );
                }
            })
        );
        this.subscription(
            'route.params',
            this._route.paramMap.subscribe(
                (params) =>
                    (this.page = params.has('view') ? params.get('view') : '')
            )
        );
    }

    public async setRoomAvailability() {
        const ref = this._dialog.open(AvailableRoomsStateModalComponent, {
            data: {
                type: 'Catering',
                disabled_rooms: await this._catering.availability
                    .pipe(take(1))
                    .toPromise(),
            },
        });
        this.subscription(
            'room-availability',
            ref.componentInstance.change.subscribe(async (list) => {
                await this._catering
                    .saveSettings({ disabled_rooms: list })
                    .catch();
                ref.componentInstance.loading = false;
            })
        );
    }

    public setChargeCodes() {
        this._dialog.open(ChargeCodeListModalComponent);
    }
}
