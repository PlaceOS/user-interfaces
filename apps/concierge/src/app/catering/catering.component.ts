import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { BaseClass } from '@placeos/common';

@Component({
    selector: '[app-catering]',
    template: `
        <sidebar></sidebar>
        <main class="relative w-full flex flex-col">
            <catering-topbar class="relative z-10"></catering-topbar>
            <ng-container [ngSwitch]="page">
                <catering-order-list
                    *ngSwitchCase="'orders'"
                    class="flex-1"
                ></catering-order-list>
                <catering-menu
                    *ngSwitchCase="'menu'"
                    class="flex-1"
                ></catering-menu>
                <div
                    *ngSwitchDefault
                    class="flex flex-1 items-center justify-center"
                >
                    <a
                        matRipple
                        class="rounded flex flex-col items-center bg-white shadow text-black"
                        [routerLink]="['/catering', 'menu']"
                    >
                        <div
                            name="img"
                            class="relative w-full flex flex-1 items-center justify-center text-2xl text-white bg-center bg-cover"
                            [style.background-image]="'url(assets/menus.jpg)'"
                        >
                            <div
                                class="absolute inset-0 bg-black opacity-60 z-0"
                            ></div>
                            <span class="z-10">Menus and Pricing</span>
                        </div>
                        <div
                            class="p-2 h-14 w-full text-sm text-center flex items-center justify-center"
                        >
                            View and Edit Menus and Pricing
                        </div>
                    </a>
                    <a
                        matRipple
                        class="rounded flex flex-col items-center bg-white shadow text-black"
                        [routerLink]="['/catering', 'orders']"
                    >
                        <div
                            name="img"
                            class="relative w-full flex flex-1 items-center justify-center text-2xl text-white bg-center bg-cover"
                            [style.background-image]="'url(assets/orders.jpg)'"
                        >
                            <div
                                class="absolute inset-0 bg-black opacity-60 z-0"
                            ></div>
                            <span class="z-10">Today's Orders</span>
                        </div>
                        <div
                            class="p-2 h-14 w-full text-sm text-center flex items-center justify-center"
                        >
                            View Catering Orders and their statuses upon arrival
                        </div>
                    </a>
                </div>
            </ng-container>
        </main>
    `,
    styles: [
        `
            :host {
                display: flex;
                height: 100%;
                width: 100%;
                background: var(--bg);
            }

            a {
                width: 28rem;
                height: 16rem;
                margin-left: 0.5rem;
                text-decoration: none;
                transition: background 200ms;
            }

            a:hover {
                background-color: #00000001 !important;
            }

            a:first-child {
                margin: 0;
            }
        `,
    ],
})
export class CateringComponent extends BaseClass implements OnInit {
    /** Page being displayed */
    public page: string;

    constructor(private _route: ActivatedRoute) {
        super();
    }

    public ngOnInit() {
        this.subscription(
            'route.params',
            this._route.paramMap.subscribe(
                (params) =>
                    (this.page = params.has('view') ? params.get('view') : '')
            )
        );
    }
}
