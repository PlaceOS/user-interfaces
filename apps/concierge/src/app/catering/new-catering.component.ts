import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { AsyncHandler } from '@placeos/common';

@Component({
    selector: '[app-new-catering]',
    template: `
        <app-topbar></app-topbar>
        <div class="flex flex-1 h-px">
            <app-sidebar></app-sidebar>
            <main class="flex flex-col flex-1 w-1/2 h-full">
                <catering-topbar class="relative z-10"></catering-topbar>
                <div class="flex flex-col px-8 flex-1 h-1/2">
                    <div
                        class="p-2 flex items-center justify-center bg-info text-white text-sm mb-4 rounded"
                        *ngIf="page === 'menu'"
                    >
                        To edit the menu de-select all levels from the top left
                    </div>
                    <div class="flex flex-1 h-1/2 w-full overflow-auto">
                        <ng-container [ngSwitch]="page">
                            <catering-order-list
                                *ngSwitchCase="'orders'"
                                class="flex-1"
                            ></catering-order-list>
                            <catering-menu
                                *ngSwitchCase="'menu'"
                                class="flex-1"
                            ></catering-menu>
                            <div
                                *ngSwitchDefault
                                class="flex flex-1 flex-wrap items-center justify-center"
                            >
                                <a
                                    matRipple
                                    class="rounded flex flex-col items-center bg-base-100 shadow text-black"
                                    [routerLink]="['/catering', 'menu']"
                                >
                                    <div
                                        name="img"
                                        class="relative w-full flex flex-1 items-center justify-center text-2xl text-white bg-center bg-cover"
                                        [style.background-image]="
                                            'url(assets/menus.jpg)'
                                        "
                                    >
                                        <div
                                            class="absolute inset-0 bg-neutral opacity-60 z-0"
                                        ></div>
                                        <span class="z-10"
                                            >Menus and Pricing</span
                                        >
                                    </div>
                                    <div
                                        class="p-2 h-14 w-full text-sm text-center flex items-center justify-center"
                                    >
                                        View and Edit Menus and Pricing
                                    </div>
                                </a>
                                <a
                                    matRipple
                                    class="rounded flex flex-col items-center bg-base-100 shadow text-black"
                                    [routerLink]="['/catering', 'orders']"
                                >
                                    <div
                                        name="img"
                                        class="relative w-full flex flex-1 items-center justify-center text-2xl text-white bg-center bg-cover"
                                        [style.background-image]="
                                            'url(assets/orders.jpg)'
                                        "
                                    >
                                        <div
                                            class="absolute inset-0 bg-neutral opacity-60 z-0"
                                        ></div>
                                        <span class="z-10">Today's Orders</span>
                                    </div>
                                    <div
                                        class="p-2 h-14 w-full text-sm text-center flex items-center justify-center"
                                    >
                                        View Catering Orders and their statuses
                                        upon arrival
                                    </div>
                                </a>
                            </div>
                        </ng-container>
                    </div>
                </div>
            </main>
        </div>
    `,
    styles: [
        `
            :host {
                display: flex;
                flex-direction: column;
                height: 100%;
                width: 100%;
                background-color: var(--b1);
            }

            a {
                width: 28rem;
                height: 16rem;
                margin-left: 0.5rem;
                text-decoration: none;
                transition: background 200ms;
            }

            a:hover {
                opacity: 0.8;
            }

            a:first-child {
                margin: 0;
            }
        `,
    ],
})
export class NewCateringComponent extends AsyncHandler implements OnInit {
    /** Page being displayed */
    public page: string;

    constructor(private _route: ActivatedRoute) {
        super();
    }

    public ngOnInit() {
        this.subscription(
            'route.params',
            this._route.paramMap.subscribe(
                (params) =>
                    (this.page = params.has('view') ? params.get('view') : '')
            )
        );
    }
}
