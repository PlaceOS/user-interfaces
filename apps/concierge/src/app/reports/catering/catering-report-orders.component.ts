import { Component } from '@angular/core';
import { CateringReportStateService } from './catering-report-state.service';

@Component({
    selector: 'catering-report-orders',
    template: `
        <div
            class="w-[64rem] max-w-[calc(100%-2rem)] rounded overflow-hidden bg-white shadow mx-auto my-2"
        >
            <div
                class="border-b border-gray-200 flex items-center justify-between px-4"
            >
                <h2 class="py-2 text-xl font-medium">Orders</h2>
                <button
                    mat-button-icon
                    (click)="download()"
                    matTooltip="Download list"
                >
                    <app-icon>download</app-icon>
                </button>
            </div>
            <custom-table
                class="w-full h-full"
                [dataSource]="orders"
                [columns]="['deliver_at', 'item_count', 'total_cost']"
                [display_column]="['Date', 'Items', 'Total Cost']"
                [column_size]="['flex']"
                [template]="{
                    deliver_at: date_state,
                    total_cost: cost_state
                }"
                [pagination]="true"
                empty="No orders for selected period"
            ></custom-table>
            <ng-template #date_state let-data="data">
                {{ data | date }} at {{ data | date: 'shortTime' }}
            </ng-template>
            <ng-template #cost_state let-data="data">
                {{ data / 100 | currency }}
            </ng-template>
        </div>
    `,
    styles: [``],
})
export class CateringReportOrdersComponent {
    public readonly orders = this._report.catering_orders;

    constructor(private _report: CateringReportStateService) {}

    public readonly download = () => this._report.downloadOrders();
}
