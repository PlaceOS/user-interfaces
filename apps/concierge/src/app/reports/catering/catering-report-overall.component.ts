import { Component } from '@angular/core';
import { CateringReportStateService } from './catering-report-state.service';

@Component({
    selector: 'catering-report-overall',
    template: `
        <div
            class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
        >
            <div class="text-3xl">
                {{ (stats | async)?.order_count || '0' }}
            </div>
            <div class="">Orders</div>
        </div>
        <div
            class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
        >
            <div class="text-3xl">
                {{ (stats | async)?.unique_items || '0' }}
            </div>
            <div class="">Unique Items</div>
        </div>
        <div
            class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
        >
            <div class="text-3xl">{{ (stats | async)?.item_count || '0' }}</div>
            <div class="">Ordered Items</div>
        </div>
        <div
            class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
        >
            <div class="text-3xl">
                {{ (stats | async)?.total_cost / 100 || 0 | currency }}
            </div>
            <div class="">Total</div>
        </div>
        <div
            class="bg-white shadow rounded m-2 px-8 w-48 h-24 flex flex-col items-center justify-center"
        >
            <div class="text-3xl">
                {{ (stats | async)?.avg_cost / 100 || 0 | currency }}
            </div>
            <div class="">Order Average</div>
        </div>
    `,
    styles: [
        `
            :host {
                display: flex;
                flex-wrap: wrap;
                align-items: center;
                justify-content: center;
                padding: 0.5rem;
            }
        `,
    ],
})
export class CateringReportOverallComponent {
    public readonly stats = this._report.stats;

    constructor(private _report: CateringReportStateService) {}
}
