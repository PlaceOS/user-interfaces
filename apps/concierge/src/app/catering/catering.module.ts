import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { RouterModule, Route } from '@angular/router';

import { MatTabsModule } from '@angular/material/tabs';
import { MatChipsModule } from '@angular/material/chips';

import { UIModule } from '../ui/ui.module';

import { CateringComponent } from './catering.component';
import { CateringTopbarComponent } from './catering-topbar.component';
import { SharedCateringModule } from '@placeos/catering';

const ROUTES: Route[] = [
    { path: '', component: CateringComponent },
    { path: ':view', component: CateringComponent },
];

@NgModule({
    declarations: [CateringComponent, CateringTopbarComponent],
    imports: [
        CommonModule,
        FormsModule,
        UIModule,
        MatTabsModule,
        MatChipsModule,
        SharedCateringModule,
        RouterModule.forChild(ROUTES),
    ],
})
export class CateringModule {}
