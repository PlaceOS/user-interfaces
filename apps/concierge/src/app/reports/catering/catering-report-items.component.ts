import { Component } from '@angular/core';
import { CateringOption } from '@placeos/catering';
import { CateringReportStateService } from './catering-report-state.service';

@Component({
    selector: 'catering-report-items',
    template: `
        <div
            class="w-[64rem] max-w-[calc(100%-2rem)] rounded overflow-hidden bg-white shadow mx-auto my-2"
        >
            <div
                class="border-b border-gray-200 flex items-center justify-between px-4"
            >
                <h2 class="py-2 text-xl font-medium">Ordered Items</h2>
            </div>
            <custom-table
                class="w-full h-full"
                [dataSource]="items"
                [columns]="[
                    'name',
                    'options',
                    'quantity',
                    'unit_price',
                    'total_cost'
                ]"
                [display_column]="[
                    'Name',
                    'Options',
                    'Quantity',
                    'Unit Price',
                    'Total Cost'
                ]"
                [column_size]="['flex']"
                [template]="{
                    options: option_state,
                    unit_price: cost_state,
                    total_cost: cost_state
                }"
                [pagination]="true"
                empty="No orders for selected period"
            ></custom-table>
            <ng-template #option_state let-data="data">
                <span
                    class="text-xs px-2 py-1 rounded bg-gray-300"
                    *ngIf="data.length"
                    [matTooltip]="options(data)"
                >
                    {{ data.length }} option(s)
                </span>
            </ng-template>
            <ng-template #cost_state let-data="data">
                {{ (data || 0) / 100 | currency }}
            </ng-template>
        </div>
    `,
    styles: [``],
})
export class CateringReportItemsComponent {
    public readonly items = this._report.catering_items;

    constructor(private _report: CateringReportStateService) {}

    public options(opts: CateringOption[]) {
        return opts.map((_) => _.name).join('\n');
    }
}
