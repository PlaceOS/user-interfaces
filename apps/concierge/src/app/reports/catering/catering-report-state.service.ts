import { Injectable } from '@angular/core';
import { CateringItem, CateringOrder } from '@placeos/catering';
import { downloadFile, flatten, jsonToCsv } from '@placeos/common';
import { CalendarEvent } from '@placeos/events';
import { combineLatest } from 'rxjs';
import { map, shareReplay, take } from 'rxjs/operators';
import { ReportsStateService } from '../reports-state.service';

@Injectable({
    providedIn: 'root',
})
export class CateringReportStateService {
    /** List of catering orders for the selected period and levels */
    public readonly catering_orders = this._reports.bookings.pipe(
        map((list) => {
            const orders: CateringOrder[] = flatten(
                list
                    .filter(
                        (_: CalendarEvent) => _.extension_data?.catering?.length
                    )
                    .map((_) =>
                        _.extension_data?.catering.map(
                            (o) =>
                                new CateringOrder({
                                    ...o,
                                    deliver_at: _.date,
                                    event: _,
                                })
                        )
                    )
            );
            console.log('Orders:', orders);
            return orders
                .filter((_) => _)
                .sort((a, b) => a.event?.date - b.event?.date);
        }),
        shareReplay(1)
    );

    /** List of catering items for the selected period and levels */
    public readonly catering_items = this.catering_orders.pipe(
        map((orders) => {
            const items: CateringItem[] = [];
            for (const order of orders) {
                for (const item of order?.items || []) {
                    const index = items.findIndex(
                        (_) => item.options_string === _.options_string
                    );
                    if (index >= 0) {
                        item[index] = new CateringItem({
                            ...item,
                            quantity: item?.quantity + item[index]?.quantity,
                        });
                    } else {
                        items.push(item);
                    }
                }
            }
            return items.sort((a, b) => b.quantity - a.quantity);
        })
    );

    public readonly stats = combineLatest([
        this.catering_orders,
        this.catering_items,
    ]).pipe(
        map(([orders, items]) => {
            const total_cost = orders.reduce((c, i) => c + i.total_cost, 0);
            return {
                order_count: orders.length,
                unique_items: items.length,
                item_count: orders.reduce((c, i) => c + i.item_count, 0),
                total_cost,
                avg_cost: total_cost / orders.length,
            };
        })
    );

    constructor(private _reports: ReportsStateService) {}

    public async downloadOrders() {
        const orders = await this.catering_orders.pipe(take(1)).toPromise();
        const csv_data = jsonToCsv(orders);
        return downloadFile('catering-orders.csv', csv_data);
    }
}
