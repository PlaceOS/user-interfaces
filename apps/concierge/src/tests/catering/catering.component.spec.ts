import { MatDialog } from '@angular/material/dialog';
import { createRoutingFactory, SpectatorRouting } from '@ngneat/spectator/jest';
import {
    CateringMenuComponent,
    CateringOrderListComponent,
} from '@placeos/catering';
import { MockComponent } from 'ng-mocks';
import { CateringTopbarComponent } from '../../app/catering/catering-topbar.component';

import { CateringComponent } from '../../app/catering/catering.component';
import { SidebarComponent } from '../../app/ui/sidebar.component';

describe('CateringComponent', () => {
    let spectator: SpectatorRouting<CateringComponent>;
    const createComponent = createRoutingFactory({
        component: CateringComponent,
        providers: [{ provide: MatDialog, useValue: { open: jest.fn() } }],
        declarations: [
            MockComponent(CateringTopbarComponent),
            MockComponent(SidebarComponent),
            MockComponent(CateringMenuComponent),
            MockComponent(CateringOrderListComponent),
        ],
    });

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(spectator.element).toMatchSnapshot();
    });

    it('should handle route parameters', () => {
        expect(spectator.component.page).toBeFalsy();
        spectator.setRouteParam('view', 'orders');
        spectator.detectChanges();
        expect(spectator.component.page).toBe('orders');
    });
});
