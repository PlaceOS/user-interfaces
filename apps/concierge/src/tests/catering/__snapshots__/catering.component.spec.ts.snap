// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CateringComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <sidebar />
  <main
    class="relative w-full flex flex-col"
  >
    <catering-topbar
      class="relative z-10"
    />
    
    
    <div
      class="flex flex-1 items-center justify-center"
    >
      <a
        class="rounded flex flex-col items-center bg-white shadow text-black"
        matripple=""
      >
        <div
          class="relative w-full flex flex-1 items-center justify-center text-2xl text-white bg-center bg-cover"
          name="img"
          style="background-image: url(assets/menus.jpg);"
        >
          <div
            class="absolute inset-0 bg-black opacity-60 z-0"
          />
          <span
            class="z-10"
          >
            Menus and Pricing
          </span>
        </div>
        <div
          class="p-2 h-14 w-full text-sm text-center flex items-center justify-center"
        >
           View and Edit Menus and Pricing 
        </div>
      </a>
      <a
        class="rounded flex flex-col items-center bg-white shadow text-black"
        matripple=""
      >
        <div
          class="relative w-full flex flex-1 items-center justify-center text-2xl text-white bg-center bg-cover"
          name="img"
          style="background-image: url(assets/orders.jpg);"
        >
          <div
            class="absolute inset-0 bg-black opacity-60 z-0"
          />
          <span
            class="z-10"
          >
            Today's Orders
          </span>
        </div>
        <div
          class="p-2 h-14 w-full text-sm text-center flex items-center justify-center"
        >
           View Catering Orders and their statuses upon arrival 
        </div>
      </a>
    </div>
    
    
  </main>
</div>
`;
